Formula/xcode-kotlin.rb

```diff
@@ -0,0 +1,56 @@
+class XcodeKotlin < Formula
+  desc "Kotlin Native Xcode Plugin"
+  homepage "https://github.com/touchlab/xcode-kotlin"
+  url "https://github.com/touchlab/xcode-kotlin.git", tag: "1.2.0", revision: "3af7e5d921c3756a3adc83ad504ac0db2a5643e8"
+  license "Apache-2.0"
+  head "https://github.com/touchlab/xcode-kotlin.git", branch: "main"
+  depends_on "openjdk" => :build
+  depends_on xcode: ["12.5", :build]
+  depends_on :macos
+
+  def install
+    ohai "Deciding which architecture to build"
+    arch = `uname -m`.strip
+
+    case arch
+    when "arm64"
+      build_task = "linkReleaseExecutableMacosArm64"
+      build_dir = "macosArm64"
+    when "x86_64"
+      build_task = "linkReleaseExecutableMacosX64"
+      build_dir = "macosX64"
+    else
+      odie "Unsupported macOS architecture #{arch}."
+    end
+    ohai "Building executable for #{arch}"
+    build_status = run_shell "./gradlew --no-daemon #{build_task} preparePlugin"
+    odie "Build FAILED" unless build_status.success?
```

Comment:
```suggestion
    system bin/"gradle", "--no-daemon", build_task, "preparePlugin"
```

Reply:
Can I use the `./gradlew` or do I have to use `bin/"gradle"`?

Reply:
It should use the formula

Reply:
I'm sorry I'm a bit confused. When you say "the formula", you mean "gradle" formula? It's important we use the right Gradle version and `./gradlew` ensures the correct version is being used. It's also the command that builds the `xcode-kotlin` binary which is then moved to `bin` and `gradle` is no longer being used afterwards.

Reply:
> I'm sorry I'm a bit confused. When you say "the formula", you mean "gradle" formula? It's important we use the right Gradle version and ./gradlew ensures the correct version is being used. 

Yes, I do mean the gradle formula. Homebrew Formula should not download their own dependencies even if they are only used at build time. If it doesn't build with the latest gradle that should be reported upstream and fixed.

Reply:
Trying to run it with `bin/"gradle"` didn't work, but just `"gradle"` does (when I replaced `openjdk` with `gradle` in `depends_on` that is). Is that a correct way to invoke it, or should it be different?